source/common/router/config_impl.cc

```diff
@@ -467,6 +467,20 @@ VirtualHostImpl::VirtualClusterEntry::VirtualClusterEntry(const Json::Object& vi
   priority_ = ConfigUtility::parsePriority(virtual_cluster);
 }
 
+const VirtualHostSharedPtr RouteMatcher::findWildcardVirtualHost(const std::string& host) const {
+  for (auto& iter : wildcard_virtual_host_suffixes_) {
+    uint32_t i = iter.first;
+    const auto& wildcard_map = iter.second;
+    // >= because *.foo.com shouldn't match .foo.com.
+    if (i >= host.size() ||
+        wildcard_map.find(host.substr(host.size() - i)) == wildcard_map.end()) {
+      continue;
+    }
+    return wildcard_map.find(host.substr(host.size() - i))->second;
```

Comment:
I'd probably write this conditional the other way around: the "found" case is what's in the if condition and returns, and the continue is implied.  I think it's a bit more clear because you affirmatively highlight the condition you're looking for.